pkgs/os-specific/linux/minimal-bootstrap/stage0-posix/make-bootstrap-sources.nix

```diff
@@ -0,0 +1,46 @@
+# Packaged resources required for the first bootstrapping stage.
+# Contains source code and 256-byte hex0 binary seed.
+#
+# We don't have access to utilities such as fetchgit and fetchzip since this
+# is this is part of the bootstrap process and would introduce a circular
+# dependency. The only tool we have to fetch source trees is `import <nix/fetchurl.nix>`
+# with the unpack option, taking a NAR file as input. This requires source
+# tarballs to be repackaged.
+#
+# To build:
+#
+#   nix-build pkgs/os-specific/linux/minimal-bootstrap/stage0-posix/make-bootstrap-sources.nix
+#   => ./result/stage0-posix-$version-$rev-source.nar.xz
+#
+
+{ pkgs ? import ../../../../.. {} }:
```

Comment:
Just so people don't cry foul you might want to instead pass the result of this expression (a cleaned-up, all-source-code tarball) as a top-level argument to `minimal-bootstrap` -- like you do for `lib`, `hostPlatform`, and `buildPlatform`.

If you do the same thing for `hex0-seed` and all the other tarballs you can even eliminate use of the Nix builtin fetchers, which would be an added win.  You could wrap them up in a single `src` attrset which becomes the top-level argument.

Anybody who sits down to convince themselves that this does what it says will start by looking at the top-level arguments and auditing those to make sure they count as "source code" (which they do).  Then from that point on it's a closed system with no builtin fetchers or anything like that, so the reviewer can shift gears from looking for loopholes to trying to understand things.